internal/core/metadata/rest_addressable.go

```diff
@@ -51,31 +53,29 @@ func restAddAddressable(w http.ResponseWriter, r *http.Request) {
 	var a models.Addressable
 	err := json.NewDecoder(r.Body).Decode(&a)
 	if err != nil {
-		LoggingClient.Error(err.Error(), "")
+		LoggingClient.Error(err.Error())
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
-	} else {
-		if len(a.Name) == 0 {
-			err = errors.New("name is required for addressable")
-			LoggingClient.Error(err.Error(), "")
-			http.Error(w, err.Error(), http.StatusBadRequest)
-			return
-		}
 	}
-	var id bson.ObjectId
-	id, err = dbClient.AddAddressable(&a)
+
+	id, err := addAddressable(a)
 	if err != nil {
-		if err == db.ErrNotUnique {
+		switch err.(type) {
+		case types.ErrDuplicateAddressableName:
 			http.Error(w, "Duplicate name for addressable", http.StatusConflict)
```

Comment:
The string can be replaced by `err.Error()`. This should be done also in the other case.

Reply:
Thank you!